### PR TITLE
Proper sentence-transformers ONNX export support

### PR DESCRIPTION
**Pull request by fxmarty**
As reported in https://github.com/huggingface/optimum/issues/1519, simple mapping `sentence-transformers` to `transformers` library allows to use only a subset of `sentence-transformers` library.

This PR adds the support of the export of `sentence_embedding` for sentence-transformers models.

Examples:
```
optimum-cli export onnx -m sentence-transformers/clip-ViT-B-32-multilingual-v1 clip_vit_multilingual_onnx
optimum-cli export onnx -m sentence-transformers/all-MiniLM-L6-v2 minilm_onnx
```

### CODE REVIEW

**Comment by JingyaHuang** [comment]
Can those extra outputs be supported directly in transformers? Just find the changes a bit hacky, and this is causing errors in optimum neuron subpackage: https://github.com/aws-neuron/aws-neuron-sdk/issues/808

**Comment by JingyaHuang** [comment]
standardizing model attributes made it a bit misleading for debug, eg. it was surprising to get `sentence-transformers-transformer` as `model_type` from a config where the `model_type` is marked as `bert`:
image

**Comment by fxmarty** [comment]
I kind of doubt this is feasible, as sentence-transformers adds quite a few features on top of transformers. For example, the sentence embeddings.

File: tests/exporters/onnx/test_exporters_onnx_cli.py [hunk 1]
```diff
@@ -204,6 +209,23 @@ def test_exporters_cli_pytorch_gpu_stable_diffusion(self, model_type: str, model
     def test_exporters_cli_fp16_stable_diffusion(self, model_type: str, model_name: str):
         self._onnx_export(model_name, model_type, device="cuda", fp16=True)
 
+    @parameterized.expand(_get_models_to_test(PYTORCH_SENTENCE_TRANSFORMERS_MODEL))
+    @require_torch
+    @require_vision
+    @require_sentence_transformers
+    @pytest.mark.timm_test
```
Comment (JingyaHuang) [inline tests/exporters/onnx/test_exporters_onnx_cli.py:216]: Why is it timm test? @fxmarty

Comment (fxmarty) [reply tests/exporters/onnx/test_exporters_onnx_cli.py:216]: It is a typo

**Pull request merged**